src/validations/passport-weighted/index.ts

```diff
@@ -14,35 +24,75 @@ export default class extends Validation {
-      passport,
-      this.author,
-      this.params.stamps
-    );
-    if (!verifiedStamps.length) return false;
-
-    const provider = snapshot.utils.getProvider(this.network);
-    const proposalTs = (await provider.getBlock(this.snapshot)).timestamp;
-
-    let weight = 0;
-    this.params.stamps.forEach((stamp: any) => {
-      const verifiedStamp = verifiedStamps.find(
-        (s: any) => s.provider === stamp.id
-      );
-
-      // check that the credential is still valid (created before snapshot block and not expired)
-      if (
-        verifiedStamp &&
-        hasValidIssuanceAndExpiration(verifiedStamp.credential, proposalTs)
-      )
-        weight += stamp.weight;
-    });
-
-    return weight >= this.params.min_weight;
+  async getSigningMessage() {
+    try {
+        const response = await fetch(SIGNING_MESSAGE_URI, {
+        headers
+        })
+        const json = await response.json()
+        return json
+    } catch (err) {
+        console.log('error: ', err)
+    }
+  }
+
+  async submitPassport() {
+    try {
+      const { message, nonce } = await this.getSigningMessage()
+
+      const provider = snapshot.utils.getProvider(this.network);
+      const signer = await provider.getSigner()
+
+      let signature = '';
+      try {
+        signature = await signer.signMessage(message);
+      } catch (err) {
+        console.log('Error signing message:', err);
+      }
+      
+      const address = this.author;
+
+      const response = await fetch(SUBMIT_PASSPORT_URI, {
+        method: 'POST',
+        headers,
+        body: JSON.stringify({
+          address,
+          scorer_id: SCORER_ID,
+          signature,
+          nonce
+        })
+      })
+      const data = await response.json()
+      console.log('data:', data)
+    } catch (err) {
+      console.log('error: ', err)
+    }
+  }
+
+  async validate(currentAddress = this.author): Promise<boolean> {
+      const GET_PASSPORT_SCORE_URI = `https://api.scorer.gitcoin.co/registry/score/${SCORER_ID}/${currentAddress}`
+      await this.getSigningMessage();
+      await this.submitPassport();
+      
+      const THRESHOLD_NUMBER = this.params.minScore || 1;
+
+      try {
+        const response = await fetch(GET_PASSPORT_SCORE_URI, {
+          headers
+        })
+        const passportData = await response.json()
+        if (passportData.score) {
+          const roundedScore = Math.round(passportData.score * 100) / 100
+          if (roundedScore >= THRESHOLD_NUMBER) return true;
+          console.log(`Your passport score (${roundedScore}) is lower than the threshold score (${THRESHOLD_NUMBER}).`);
+          return false;
+        } else {
+          console.log('You do not have a valid Gitcoin Passport. Create one by visiting https://passport.gitcoin.co/ ')
+          return false;
+          }
+      } catch (err) {
+        console.log('error: ', err)
+        return false;
+      }
```

Comment:
in case of an error, if we return false it can cause confusion, so can remove try-catch, we handle it somewhere else

Reply:
Made a new commit that changed return false to throw err so that if an error occurs, it won't be mistaken for a validation failure. This code will log the error as before, but then it will rethrow it to allow the error to propagate up the call stack to be handled by a global error handler.